builtin/logical/database/path_config_connection.go

```diff
@@ -217,7 +218,7 @@ func (b *databaseBackend) connectionDeleteHandler() framework.OperationFunc {
 
 		err := req.Storage.Delete(ctx, fmt.Sprintf("config/%s", name))
 		if err != nil {
-			return nil, errors.New("failed to delete connection configuration")
+			return nil, errwrap.Wrapf("failed to delete connection configuration: {{err}}", err)
```

Comment:
How does this fix the deletion issue? Was this swallowing a read-only error that was needed to forward the request?